fiftyone/core/odm/mixins.py

```
@@ -141,16 +142,35 @@ def set_field(
+                self._changed_fields.remove(root)
 
             # handle sytnax: sample["field.0.attr"] = value
             if isinstance(doc, (mongoengine.base.BaseList, fof.ListField)):
                 chunks = chunks[1].split(".", 1)
                 doc = doc[int(chunks[0])]
+                field_name = root + "." + chunks[1]
 
-            return doc.set_field(chunks[1], value, create=create)
+            doc.set_field(chunks[1], value, create=create)
+
+            #
```

Ruff (0.8.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)